platforms/axplat-aarch64-bsta1000b/src/dw_apb_uart.rs

```diff
@@ -0,0 +1,79 @@
+//! snps,dw-apb-uart serial driver
+
+use arm_gicv2::{InterruptType, translate_irq};
+use axplat::console::ConsoleIf;
+
+use crate::{config::devices::UART_IRQ, mem::phys_to_virt};
+use dw_apb_uart::DW8250;
+use kspin::SpinNoIrq;
+use memory_addr::PhysAddr;
+
+const UART_BASE: PhysAddr = pa!(crate::config::devices::UART_PADDR);
+
+static UART: SpinNoIrq<DW8250> = SpinNoIrq::new(DW8250::new(phys_to_virt(UART_BASE).as_usize()));
+
+/// The GIC INTID for the UART interrupt.
+const UART_IRQ_NUM: usize = translate_irq(UART_IRQ, InterruptType::SPI).unwrap();
```

Comment:
We can have `UART_IRQ` add the SPI offset (0x20) in `axconfig.toml`  to remove the `arm_gicv2` dependencies.